### PR TITLE
Fixed issues around reduced format

### PR DESCRIPTION
**Pull request by pwrose**

### CODE REVIEW

File: mmtf-codec/src/main/java/org/rcsb/mmtf/encoder/ReducedEncoder.java [hunk 1]
```diff
@@ -1,225 +1,315 @@
...
+				for(int i = 0; i < structure.getNumAtomsInGroup(groupIndex); i++) {
+					atomIndices.add(i);
 				}
 			}
-		}
-		// Check if it's a non-polymer 
-		else if (chainType.equals("non-polymer")){
-			for(int i=0; i<structureDataInterface.getNumAtomsInGroup(groupType); i++){
-				outList.add(i);
+		} else if (! structure.getGroupName(groupIndex).equals("HOH")){
```
Comment (josemduarte) [inline mmtf-codec/src/main/java/org/rcsb/mmtf/encoder/ReducedEncoder.java]: How about deuterated water?

File: mmtf-codec/src/test/java/org/rcsb/mmtf/encoder/TestReducedEncoderComprehensive.java [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+import org.rcsb.mmtf.decoder.GenericDecoder;
+import org.rcsb.mmtf.decoder.ReaderUtils;
+
+public class TestReducedEncoderComprehensive {
+
+	@Test
+	public void test() throws IOException, ParseException {
+		StructureDataInterface full = getDefaultFullData();
+		StructureDataInterface reduced = ReducedEncoder.getReduced(getDefaultFullData());
+		compareFullVsReduced(full.getStructureId(), full, reduced);
```
Comment (josemduarte) [inline mmtf-codec/src/test/java/org/rcsb/mmtf/encoder/TestReducedEncoderComprehensive.java]: How about including a few more cases here, not only 4cup. E.g. structure with nucleotides, structure with water marked as non-polymer, structure with alt conformations, etc...

**Review by josemduarte — CHANGES_REQUESTED** [review]
Looks good! I've commented on a couple of issues. Specially it'd be good to have testing on other edge cases.

**Comment by josemduarte** [comment]
Thanks for the changes @pwrose ! One issue now, could you check why the tests are not passing?

By the way, I forgot to mention yesterday that a possible way to do the tests is by using https://github.com/rcsb/mmtf/tree/master/test-suite

**Comment by pwrose** [comment]
Btw, I have used the test-suite files. I've put a local copy of the files
in the resources folder so the tests can be run offline.

**Review by josemduarte — APPROVED** [review]
